api/query_engine.go

```diff
@@ -37,6 +38,27 @@ func (a archives) Len() int           { return len(a) }
 func (a archives) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a archives) Less(i, j int) bool { return a[i].interval < a[j].interval }
 
+// represents a numbered data "archive", i.e. the raw one, or an aggregated series
+// number 0 = raw archive, number 1,2,..N = aggregation 1,2..N
+// this allows you to reorder narchives and still keep track of which is which
+type narchive struct {
+	i          int
```

Comment:
To understand the purpose of this `i` it's necessary to go quite deep into how `narchive` is used. Some explanatory name or some comment why it is necessary to keep that instead of just relying on the index of the `options` list would be helpful.